lib/rack/request.rb

```diff
@@ -16,7 +16,18 @@ class << self
       attr_accessor :ip_filter
     end
 
-    self.ip_filter = lambda { |ip| /\A127\.0\.0\.1\Z|\A(10|172\.(1[6-9]|2[0-9]|30|31)|192\.168)\.|\A::1\Z|\Afd[0-9a-f]{2}:.+|\Alocalhost\Z|\Aunix\Z|\Aunix:/i.match?(ip) }
+    TRUSTED_PROXIES = Regexp.union(
+      /\A127\.(([1-9]?\d|[12]\d\d)\.){2}([1-9]?\d|[12]\d\d)\Z/i,     # localhost IPv4 range 127.x.x.x, per RFC-3330
+      /\A::1\Z/i,                                                    # localhost IPv6 ::1
+      /\A[fF][cCdD][0-9a-fA-F]{2}(?:[:][0-9a-fA-F]{0,4}){0,7}\Z/i,   # private IPv6 range fc00 .. fdff
```

Comment:
You don't need `[fF]` and similar in a case insensitive regexp.  Stick to just the lowercase or uppercase character.